server/container/core/src/main/java/org/apache/james/core/MailImpl.java

```diff
@@ -541,6 +542,7 @@ private void writeObject(java.io.ObjectOutputStream out) throws IOException {
         out.writeObject(remoteAddr);
         out.writeObject(lastUpdated);
         out.writeObject(attributes);
+        out.writeObject(perRecipientSpecificHeaders);
```

Comment:
My understanding is the previous writer wants this serialization to be as stable as possible.

They gained more stability (across class versions) by specifying manually the fields to write.

Anyway, I also checked, and this serialisation code is not used anywhere...